ngraph/test/preprocess.cpp

```diff
@@ -327,3 +330,208 @@ TEST(pre_post_process, resize_no_tensor_width) {
+            .output(OutputInfo()
+                        .network(OutputNetworkInfo().set_layout("NCHW"))
+                        .postprocess(PostProcessSteps().convert_layout())
+                        .tensor(OutputTensorInfo().set_layout("NHWC")))
+            .build(f);
+    EXPECT_EQ(f->get_results()[0]->get_layout(), "NHWC");
+    EXPECT_EQ(f->get_results()[0]->get_output_tensor(0).get_partial_shape(), (PartialShape{1, 2, 2, 3}));
+}
+
+TEST(pre_post_process, postprocess_convert_layout_same) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 2, 2});
+    auto size_old = f->get_ordered_ops().size();
+
+    f = PrePostProcessor()
+            .output(OutputInfo()
+                        .network(OutputNetworkInfo().set_layout("NCHW"))
+                        .postprocess(PostProcessSteps().convert_layout("NCHW"))
+                        .tensor(OutputTensorInfo().set_layout("NCHW")))
+            .build(f);
+    EXPECT_EQ(f->get_results()[0]->get_layout(), "NCHW");
+    EXPECT_EQ(f->get_results()[0]->get_output_tensor(0).get_partial_shape(), (PartialShape{1, 3, 2, 2}));
+    // Verify that redundant ops were not added
+    EXPECT_EQ(size_old, f->get_ordered_ops().size());
+}
+
+TEST(pre_post_process, postprocess_convert_layout_default_error) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 2, 2});
+
+    EXPECT_THROW(f = PrePostProcessor()
+                         .output(OutputInfo()
+                                     .network(OutputNetworkInfo().set_layout("NCHW"))
+                                     .postprocess(PostProcessSteps().convert_layout()))
+                         .build(f),
+                 ov::AssertFailure);
+}
+
+// Postprocessing - other
+
+TEST(pre_post_process, postprocess_custom_step) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 2, 2});
+    std::string name;
+    f = PrePostProcessor()
+            .output(OutputInfo().postprocess(
+                PostProcessSteps().custom([&name](const ov::Output<Node>& node) -> ov::Output<Node> {
+                    auto abs = std::make_shared<op::v0::Abs>(node);
+                    abs->set_friendly_name(node.get_node()->get_friendly_name() + "/abs");
+                    name = node.get_node()->get_friendly_name() + "/abs";
+                    return abs;
+                })))
+            .build(f);
+    EXPECT_FALSE(name.empty());
+    EXPECT_EQ(f->get_results()[0]->get_input_source_output(0).get_node()->get_friendly_name(), name);
+}
+
+TEST(pre_post_process, postprocess_implicit_convert_element_type_and_layout) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 2, 2});
+    f = PrePostProcessor()
+            .output(OutputInfo()
+                        .network(OutputNetworkInfo().set_layout("NCHW"))
+                        .tensor(OutputTensorInfo().set_layout("NHWC").set_element_type(element::u8)))
+            .build(f);
+    EXPECT_EQ(f->get_results()[0]->get_element_type(), element::u8);
```

Comment:
should not we check shape order as well?

Reply:
Yes, missed this, thanks, fixes